packager/media/crypto/subsample_generator.cc

```diff
@@ -297,6 +300,24 @@ Status SubsampleGenerator::GenerateSubsamplesFromVPxFrame(
   return Status::OK;
 }
 
+Status SubsampleGenerator::GenerateSubsamplesFromAC4Frame(
+    const uint8_t* frame,
+    size_t frame_size,
+    std::vector<SubsampleEntry>* subsamples) {
+  SubsampleOrganizer subsample_organizer(align_protected_data_,
+      subsamples);
+  size_t toc_size = 0;
+  AC4Parser ac4_frame;
+  if (ac4_frame.Parse(frame, frame_size)) {
+    toc_size = ac4_frame.GetAc4TocSize();
+  }
+  // cenc: cipher_bytes should be a multiple of 16 bytes.
+  size_t clear_bytes = (toc_size + 8 * ((toc_size % 8 == 0) ? 0 : 1)) / 8;
+  size_t cipher_bytes = frame_size - clear_bytes;
```

Comment:
If cipher_bytes must be a multiple of 16, as mentioned in the comment above, how do we know that is true here?  What's the logic?  Should there be a DCHECK on that?